plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/workspace/WorkspaceProjectStorageCleanerImpl.java

```diff
@@ -0,0 +1,96 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.workspace;
+
+import com.codenvy.machine.backup.WorkspaceIdHashLocationFinder;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.eclipse.che.api.core.util.CommandLine;
+import org.eclipse.che.api.core.util.ListLineConsumer;
+import org.eclipse.che.api.core.util.ProcessUtil;
+import org.eclipse.che.api.workspace.server.WorkspaceProjectStorageCleaner;
+import org.eclipse.che.commons.lang.concurrent.ThreadLocalPropagateContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeoutException;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Component to clean up workspace project storage which can be located in the remote node.
+ *
+ * @author Alexander Andrienko
+ */
+@Singleton
+public class WorkspaceProjectStorageCleanerImpl implements WorkspaceProjectStorageCleaner {
+
+    private static final Logger LOG = LoggerFactory.getLogger(WorkspaceProjectStorageCleanerImpl.class);
+
+    private final WorkspaceIdHashLocationFinder workspaceIdHashLocationFinder;
+    private final File                          backupsRootDir;
+    private final int                           cleanUpTimeOut;
+    private final String                        workspaceCleanUpScript;
+    private final ExecutorService               executor;
+
+    @Inject
+    public WorkspaceProjectStorageCleanerImpl(WorkspaceIdHashLocationFinder workspaceIdHashLocationFinder,
+                                              @Named("che.user.workspaces.storage") File backupsRootDir,
+                                              @Named("workspace.cleanup.cleanup_workspace_storage_script") String workspaceCleanUpScript,
+                                              @Named("workspace.cleanup.cleanup_duration_second") int cleanUpTimeOut) {
```

Comment:
I think that `workspace.projects_storage.cleanup.duration_second` would be better